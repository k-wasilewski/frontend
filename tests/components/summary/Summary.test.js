import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../src/redux/store";
import {Provider} from "react-redux";
import {configure, shallow} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import ConnectedSummary, { Summary } from "../../../src/components/summary/Summary"

describe("Summary specification", () => {
    it('renders a title and a button initially', () => {
        const component = renderer.create(
            <Provider store={store}>
                <ConnectedSummary />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const summary = tree.children[0]
        expect(summary.children[0].children[0]).toEqual('Podsumowanie')
        expect(summary.children[2].children[0].children[0]).toEqual('Szukaj')
    })

    it('doGetList()', () => {
        configure({ adapter: new Adapter() });

        const component = shallow(
            <Summary />
        )

        let mockClick = () => component.find('.col2')[0].children[0].simulate('click')
    })
})