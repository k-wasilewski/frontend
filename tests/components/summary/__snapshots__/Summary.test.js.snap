// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary specification renders a list and a button 1`] = `
<div
  className="main"
>
  <div
    className="summary"
  >
    <h2>
      Podsumowanie
    </h2>
    <div
      className="col1"
    >
      <p />
    </div>
    <div
      className="col2"
    >
      <button
        onClick={[Function]}
      >
        Szukaj
      </button>
    </div>
  </div>
</div>
`;

exports[`Summary specification renders a title and a button initially 1`] = `
<div
  className="main"
>
  <div
    className="summary"
  >
    <h2>
      Podsumowanie
    </h2>
    <div
      className="col1"
    >
      <p />
    </div>
    <div
      className="col2"
    >
      <button
        onClick={[Function]}
      >
        Szukaj
      </button>
    </div>
  </div>
</div>
`;
